Dockerfile

```
@@ -1,10 +1,13 @@
-FROM python:3.11
+FROM python:3.12-slim
+# Ensure poetry does not create a virtualenv
+ENV POETRY_VIRTUALENVS_CREATE=false
 
-WORKDIR /app
+RUN pip install poetry
+WORKDIR /code
+COPY poetry.lock pyproject.toml /code/
+RUN poetry install --no-interaction --no-ansi --no-root --no-dev
 
-COPY pyproject.toml .
-RUN pip install poetry && poetry install
-RUN pip install fastapi uvicorn
-
-COPY . .
-RUN chmod +x ./entrypoint.sh
\ No newline at end of file
+COPY cid /code/cid/
+COPY entrypoint.sh /code/
+RUN chmod +x /code/entrypoint.sh
```

We also need to copy import_data.py. I also noticed error messages in the container logs when README.md isn't available at poetry install time (unclear why this is necessary).

```suggestion
COPY import_data.py /code/
COPY README.md /code/
RUN chmod +x /code/entrypoint.sh
```